sn_logging/src/metrics.rs

```diff
@@ -0,0 +1,146 @@
+// Copyright 2023 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use serde::Serialize;
+use std::time::Duration;
+use sysinfo::{self, CpuExt, NetworkExt, Pid, PidExt, ProcessExt, System, SystemExt};
+use tracing::trace;
+
+const UPDATE_INTERVAL: Duration = Duration::from_secs(5);
+const TO_MB: f32 = 1_000_000.0;
+
+// The following Metrics are collected and logged
+#[derive(Debug, Serialize)]
+#[allow(dead_code)]
+struct Metrics {
+    // Number of threads in the system
+    physical_cpu_threads: usize,
+    // Percentage of CPU used
+    cpu_usage_percent: f32,
+    // Total system memory in MBytes
+    total_memory_mb: f32,
+    // RAM used in MBytes
+    memory_used_mb: f32,
+    // Percentage of RAM used
+    memory_usage_percent: f32,
+    // Network metrics is None if the default network interface cannot be found
+    network: Option<NetworkMetrics>,
+    // Process metrics is None if the Pid for the process is invalid
+    process: Option<ProcessMetrics>,
+}
+
+#[derive(Debug, Serialize)]
+#[allow(dead_code)]
+struct NetworkMetrics {
+    // The default network interface
+    interface_name: String,
+    // Bytes received during UPDATE_INTERVAL
+    bytes_received: u64,
+    // Bytes transmitted during UPDATE_INTERVAL
+    bytes_transmitted: u64,
+    // The total MBytes received through the interface
+    total_mb_received: f32,
+    // The total MBytes transmitted through the interface
+    total_mb_transmitted: f32,
+}
+
+#[derive(Debug, Serialize)]
+#[allow(dead_code)]
+struct ProcessMetrics {
+    // Percentage of CPU used by the process
+    cpu_usage_percent: f32,
```

Comment:
is this somehow overlapped with the fields within `Metrics` ?
maybe just make 
```rust
struct Metrics {
    network_metrics: NetworkMetrics,
    process_metrics: ProcessMetrics,
}
```

Reply:
I have renamed the `Metrics::cpu_usage_percent` to `Metrics::system_cpu_usage_percent` to denote that the outer ones denote system-wide usage. Whereas the `Metrics::process::*` corresponds to `safe` or `safenode` bin depending on what we run.